### DISCUSSION TITLE
二进制文件的处理方式？

### DISCUSSION THREAD

**Discussion by Wenti-D** [Physics365/991CN-X-CW-Decompilation#19] [General]
众所周知，现在仓库里的二进制文件已经超过 100M，但众不所周知，如果不花钱的话，仓库每个周期（不知道是多久）只有 1G 的 LFS 流量，所以……

那么这些个二进制文件到底怎么办呢？

**Comment by Physics365** [comment]
完了，我昨天不小心把本地文件删了，然后现在。。。
首先现在我怎么办

**Reply by Wenti-D** [reply]
啊？发生什么事了？

**Reply by Physics365** [reply]
重新clone时，Git LFS文件下不下来了：
现在怎么办呢😭😭😭
![屏幕截图 2024-04-04 120841](https://github.com/Physics365/991CN-X-CW-Decompilation/assets/154725076/8dedddf8-985f-4e2c-9bbf-c2598316e8c8)

![屏幕截图 2024-04-04 120947](https://github.com/Physics365/991CN-X-CW-Decompilation/assets/154725076/7992fd2d-08d6-4511-bc6c-21b0f4f92a53)

**Reply by Wenti-D** [reply]
带宽满了，那就只能等重置了😂

**Comment by Physics365** [comment]
看来Git LFS 也不能放太多数据

**Comment by Physics365** [comment]
还有封面的图片不能用Git LFS，否则主README.md显示不了了（这个我已修改）

**Reply by Wenti-D** [reply]
其实不是因为 LFS 显示不了，而是因为流量无了……

**Comment by Physics365** [comment]
> 众所周知，现在仓库里的二进制文件已经超过 100M，但众不所周知，如果不花钱的话，仓库每个周期（不知道是多久）只有 1G 的 LFS 流量，所以……
> 
> 那么这些个二进制文件到底怎么办呢？

现在流量真的用尽了（周期为**1个月**）

**Reply by Wenti-D** [reply]
寄！

**Reply by Physics365** [reply]
所以这个月我们只能改一下纯文本了？

**4月刚刚开始。。。**

**Comment by qiufuyu123** [comment]
我手上有台云服务器，我回头搭个类似ftp服务器的东西，然后把大家把二进制传上来

**Reply by Wenti-D** [reply]
好好好！

**Reply by Physics365** [reply]
支持👍👍👍

**Comment by qiufuyu123** [comment]
[云盘](https://casio.123a.club/)

访客不需要登录，只可查看/下载
管理员可以上传/删除
管理员账号:admin
管理员密码:casioadmin

登陆方式：右上角按钮
(管理员账号/密码过几天修改，然后私发给所有contributor)

**Reply by Physics365** [reply]
私发是指...?

**Reply by Wenti-D** [reply]
咦，我刚刚发上去的 PDF 什么的神秘消失了？

**Reply by qiufuyu123** [reply]
哦哦哦，nginx上传大小限制忘改了

**Comment by Physics365** [comment]
所以现在谁发一下LFS文件到云盘里呀，我现在什么LFS文件都没有

**Comment by qiufuyu123** [comment]
![image](https://github.com/Physics365/991CN-X-CW-Decompilation/assets/60765480/3744a956-3397-41cd-8439-091fcdc81c93)

刚刚加了个readme自动预览
现在如果检测到当前目录有readme.md文件，会自动预览在下方

**Comment by EwTE-H** [comment]
给各位一个无需账号，不限空间不限速的直链云盘
[下载地址](https://livefile.xesimg.com/programme/python_assets/bd95d938c21dd0b3f6d445f34bcdf557.exe)
@qiufuyu123 你看看这个怎么样，你那个服务器下载有点慢，要不考虑一下存储文件用这个云盘，服务器上就存个链接
附上云盘的核心源码作为参考
```python
import requests
import json
import hashlib
def upload(filename, contents):
    response = requests.get(
        url='https://code.xueersi.com/api/assets/get_oss_upload_params', 
        params={
            "scene": "offline_python_assets", 
            "md5": hashlib.md5(contents).hexdigest(), 
            "filename": filename
        }
    )
    uploadParams = json.loads(response.text)['data']
    requests.request(
        method="PUT",
        url=uploadParams['host'],
        data=contents,
        headers=uploadParams['headers']
    )
    return uploadParams['url']
```

**Reply by qiufuyu123** [reply]
我服务器下载慢是因为我的服务器架在日本，开个梯子就没啥问题了
设置登录是怕有人上去乱删东西
然后用的服务端程序是这个:[dufs](https://github.com/sigoden/dufs)
主要不想重头造轮子了

你说的这个云盘为啥域名是 学而思（xueersi.com/api...） ?
是学而思的服务吗？

这样的话算了吧，因为放自己服务器(亿速云轻量应用服务器，1核2g，续费了一年)方便/靠谱 一点

**Comment by fish4terrisa-MSDSM** [comment]
可以看下有没有rsync之类的链接吗，我这里的vps也能存一下。实际上也可以把binary扔到release里，那个不限
